add-on/src/lib/ipfs-request.js

```diff
@@ -138,14 +138,14 @@ export function createRequestModifier (getState, dnslinkResolver, ipfsPathValida
     // browser.webRequest.onBeforeRequest
     // This event is triggered when a request is about to be made, and before headers are available.
     // This is a good place to listen if you want to cancel or redirect the request.
-    onBeforeRequest (request) {
+    async onBeforeRequest (request) {
```

Comment:
:information_source:  Very happy to see this possible in Chromium :))

`async` here was possible from Firefox 52 onwards, but Chromium did not support promises for years, and we were stuck with all the blocking XHR code here.